### PR TITLE
Linux implimentation of processes monitoring

### PR DESCRIPTION
**Pull request by VytalyGorbatov**
Added Linux implimentation based on /proc pseudofilesystem. Also, added some usefull utilities and fixed minor issues.

### CODE REVIEW

File: Server/BackEndMonitoringServer/src/CProcessInfo.cpp [hunk 1]
```diff
@@ -7,10 +7,9 @@
...
 #include "CProcessInfo.h"
 
-CProcessInfo::CProcessInfo(unsigned PID, unsigned count_of_processors, 
-                   EMemoryConvertType type) :
-    m_PID(PID),
-    m_count_of_processors(count_of_processors),m_count_type(type), 
+CProcessInfo::CProcessInfo(unsigned PID,
+                           EMemoryConvertType type) :
+    m_PID(PID),m_count_type(type), 
     m_cpu_usage(0.0), m_ram_usage(00ULL), m_pagefile_usage(00ULL),
```
Comment (Chupakabra0) [inline Server/BackEndMonitoringServer/src/CProcessInfo.cpp:13]: Question out of curiosity: why `00ULL`?

File: utils/include/CNumericTypesParser.cpp [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+    for( auto sym = scientific_part_begin; sym<scientific_part_end; ++sym) {
+        if(!std::isdigit( *sym )){
+            return false;
+        }
+    }
+    return true;
+}
+
+bool CNumericTypesParser::IsIntagerNumber() const {
+    size_t i = 0;
```
Comment (Chupakabra0) [inline utils/include/CNumericTypesParser.cpp:25]: A lot of implicit casts from `int` to `size_t`

File: utils/include/StrToNumWrapper.cpp [hunk 3]
```diff
@@ -0,0 +1,69 @@
+#include "stdafx.h"
+
+#include "StrToNumWrapper.h"
+
+namespace StrToNumWrapper{
+bool TryStoi(const std::string& val, int& var){
+    try{
+        var = std::stoi(val);
+    } catch(...) {
```
Comment (Chupakabra0) [inline utils/include/StrToNumWrapper.cpp:9]: `...` hmmm, brave XD
Catch or instead of it `const std::exception&`

**Review by Chupakabra0 — CHANGES_REQUESTED** [review]
Everything very good, but exceptions catching at cast methods.
Change it with logger's macros, that have inbuilt exception catcher and it will be insane or just catch `const std::exception&`, as you want

File: Server/BackEndMonitoringServer/src/CProcessInfo.cpp (hunk 1 above)
Comment (VytalyGorbatov) [reply Server/BackEndMonitoringServer/src/CProcessInfo.cpp]: Thnx, it's typo

File: utils/include/CNumericTypesParser.cpp (hunk 2 above)
Comment (VytalyGorbatov) [reply utils/include/CNumericTypesParser.cpp:28]: Fixed

File: utils/include/StrToNumWrapper.cpp (hunk 3 above)
Comment (VytalyGorbatov) [reply utils/include/StrToNumWrapper.cpp]: D'oh, forgot to changi this. Thnx, changed to logger's macros, please, check it now)

**Review by Chupakabra0 — APPROVED** [review]
Perfect and fast, approve

**Pull request merged**